src/devices/machine/i8291a.cpp

```diff
@@ -0,0 +1,1130 @@
+// license:BSD-3-Clause
+// copyright-holders: Sven Schnelle
+/*********************************************************************
+
+ i8291a.cpp
+
+ Intel 8291A GPIB controller
+
+ *********************************************************************/
+
+#include "emu.h"
+
+
+#include "i8291a.h"
+
+#define VERBOSE 0
+
+#include "logmacro.h"
+
+#define LOG_STATE_MASK  (LOG_GENERAL << 1)
+#define LOG_STATE(...)  LOGMASKED(LOG_STATE_MASK, __VA_ARGS__)
+#define LOG_REG_MASK    (LOG_STATE_MASK << 1)
+#define LOG_REG(...)    LOGMASKED(LOG_REG_MASK, __VA_ARGS__)
+#define LOG_INT_MASK    (LOG_REG_MASK << 1)
+#define LOG_INT(...)    LOGMASKED(LOG_INT_MASK, __VA_ARGS__)
+#define LOG_CMD_MASK	(LOG_INT_MASK << 1)
+#define LOG_CMD(...)	LOGMASKED(LOG_CMD_MASK, __VA_ARGS__)
```

Comment:
You need to define the macros that can be used for `VERBOSE` before you have the dummy `VERBOSE` declaration.  Also, since `LOG_GENERAL` doesn't have a `_MASK` suffix, please don't use it on your masks, as it creates inconsistency.

Reply:
That would clash with the actual LOG_() macros - what would be the preferred name for that?

Reply:
Well, the convention that @JoakimLarsson seemed to use was to have an underscore in the mask macro, like `LOG_INT`, but not in the logging macro, like `LOGINT(...)` - @JoakimLarsson do you have any comments on the best way to deal with this?  (Essentially, `logmacro.h` is supposed to be an adoption of the kind of logging macros @JoakimLarsson was using with the major issues addressed.)